src/EnergyPlus/OutputProcessor.hh

```diff
@@ -904,50 +926,50 @@ namespace OutputProcessor {
 // *****************************************************************************
 
 void SetupOutputVariable(EnergyPlusData &state,
-                         std::string const &VariableName,           // String Name of variable (with units)
-                         OutputProcessor::Unit const &VariableUnit, // Actual units corresponding to the actual variable
-                         Real64 &ActualVariable,                    // Actual Variable, used to set up pointer
-                         std::string const &TimeStepTypeKey,        // Zone, HeatBalance=1, HVAC, System, Plant=2
-                         std::string const &VariableTypeKey,        // State, Average=1, NonState, Sum=2
-                         std::string const &KeyedValue,             // Associated Key for this variable
-                         Optional_string_const ReportFreq = _,      // Internal use -- causes reporting at this freqency
-                         Optional_string_const ResourceTypeKey = _, // Meter Resource Type (Electricity, Gas, etc)
-                         Optional_string_const EndUseKey = _,       // Meter End Use Key (Lights, Heating, Cooling, etc)
-                         Optional_string_const EndUseSubKey = _,    // Meter End Use Sub Key (General Lights, Task Lights, etc)
-                         Optional_string_const GroupKey = _,        // Meter Super Group Key (Building, System, Plant)
-                         Optional_string_const ZoneKey = _,         // Meter Zone Key (zone name)
-                         Optional_int_const ZoneMult = _,           // Zone Multiplier, defaults to 1
-                         Optional_int_const ZoneListMult = _,       // Zone List Multiplier, defaults to 1
-                         Optional_int_const indexGroupKey = _,      // Group identifier for SQL output
-                         Optional_string_const customUnitName = _   // the custom name for the units from EMS definition of units
+                         std::string const &VariableName,                         // String Name of variable (with units)
+                         OutputProcessor::Unit const &VariableUnit,               // Actual units corresponding to the actual variable
```

Comment:
Don't pass `enum`s by reference unless you are actually returning a value through this parameter. Generally, the only parameters you should pass by reference are those you are using to return a value or objects that you don't want to copy from the caller to the callee (pass these by const reference). Pass everything else by value. Passing by reference unnecessarily is slower and also impedes compiler optimization.

Reply:
@amirroth Thanks for looking it over and I hear you.  I have to clean up a compiler warning so I'll make this change then.  

In general though, on this second attempt of this effort I was just trying to be as minimalistic as I could to avoid causing any diffs and holding this up further, fully understanding that there will be a pass that collects the enums into a smaller list and improves it in other ways as well.